atomic_reactor/plugins/pre_resolve_composes.py

```diff
@@ -362,7 +363,11 @@ def render_packages_request(self):
             'sigkeys': self.signing_intent['keys'],
         }
         if self.arches:
-            request['arches'] = self.arches
+            arches = self.arches
```

Comment:
You probably wanna create a copy here.

Reply:
Also, we need similar change in `render_pulp_request`.

Reply:
render pulp request doesn't assign anything that would need a copy afaict.
```
    def render_pulp_request(self, arch):
        return {
            'source_type': 'pulp',
            'source': ' '.join(self.pulp.get(arch, [])),
            'sigkeys': [],
            'flags': self.flags,
            'arches': [arch]
        }
```
copy what?

Reply:
When you do this: `arches = self.arches` it doesn't create a copy, so all the changes you're doing in arches, will propagate to `self.arches`. That's error prone. Let's not modify `self.arches` in this method.

In render_pulp_request, you need to insert `i686` in list of arches if arch is `x86_64`:
```python
    def render_pulp_request(self, arch):
        arches = [arch]
        if self.multilib and arch == 'x86_64':
          arches.append('i686')
        return {
            'source_type': 'pulp',
            'source': ' '.join(self.pulp.get(arch, [])),
            'sigkeys': [],
            'flags': self.flags,
            'arches': arches
        }
```

Reply:
Fixed.